packages/stores/src/chain.ts

```diff
@@ -13,7 +13,11 @@ export interface ChainInfoWithExplorer extends ChainInfo {
   /** Add optional stable coin peg info to currencies. */
   currencies: Array<
     AppCurrency & {
-      pegMechanism?: "collateralized" | "algorithmic" | "hybrid";
+      pegMechanism?:
+        | "fiat-backed"
+        | "algorithmic"
+        | "crypto-backed"
+        | "hybrid-pegged";
```

Comment:
It's nice we get more granularity in the various mechanisms.

Would you mind defining each of these types please? Perhaps in a comment as well?

Reply:
Fiat-backed: Backed by a reserve of traditional fiat currencies, such as the USD/Euro/GBP etc, providing steadiness with a 1:1 peg and reduced volatility.

Algorithmic: Leverages algorithms and contracts to adapt supply based on demand, striving to stabilize value amidst market changes.

Crypto-backed: These are collateralized by other cryptocurrencies, ensuring stability through locked assets while enabling issuance of stablecoins.

Hybrid-pegged: Mixes collateral assets such as fiat and cryptocurrencies while also utilizing algorithmic mechanisms to maintain its stability.